edge/pkg/metamanager/client/event.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2024 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	appcorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/message"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/modules"
+)
+
+type EventsGetter interface {
+	Events(namespace string) EventsInterface
+}
+
+type EventsInterface interface {
+	Create(*corev1.Event, metav1.CreateOptions) (*corev1.Event, error)
+	Update(*corev1.Event, metav1.UpdateOptions) (*corev1.Event, error)
+	Patch(string, types.PatchType, []byte, metav1.PatchOptions) (*corev1.Event, error)
+	Delete(string, metav1.DeleteOptions) error
+	Get(string, metav1.GetOptions) (*corev1.Event, error)
+	Apply(*appcorev1.EventApplyConfiguration, metav1.ApplyOptions) (result *corev1.Event, err error)
+
+	CreateWithEventNamespace(*corev1.Event) (*corev1.Event, error)
+	UpdateWithEventNamespace(*corev1.Event) (*corev1.Event, error)
+	PatchWithEventNamespace(*corev1.Event, []byte) (*corev1.Event, error)
+}
+
+type events struct {
+	send      SendInterface
+	namespace string
+}
+
+func newEvents(namespace string, s SendInterface) *events {
+	return &events{
+		send:      s,
+		namespace: namespace,
+	}
+}
+
+func (e *events) Create(event *corev1.Event, opts metav1.CreateOptions) (*corev1.Event, error) {
```

Comment:
Why doesn't the Create function do anything?

If the PR isn't finished or tested, adding 'WIP' to the title might better indicate its incomplete status

Reply:
Seems kubelet doesn't use create/update/... functions to report event, but we do need to define these function in `EventsInterface` otherwise we cannot use metaclient to replace kubeclient successfully.  So we don't need implement it but just return nil in these functions.

Reply:
Ok, 

but why not print error and  return nil when call the functiom. this helps in finding unknown callers.

Reply:
I think both approaches are acceptable. `Create` func is currently not part of the features supported by edged, so this  func will not be called.